packages/flutter/test/cupertino/route_test.dart

```diff
@@ -652,6 +658,211 @@ void main() {
+
+      expect(find.text('Page 2'), findsOneWidget);
+
+      tester.state<NavigatorState>(find.byType(Navigator)).pop();
+      await tester.pump(const Duration(milliseconds: 100));
+
+      expect(find.text('Page 2'), findsOneWidget); // Transition still in progress
+
+      await tester.tap(find.text('Home'), warnIfMissed: false); // Home route is not tappable
+      expect(homeTapped, false);
+    });
+
+    testWidgets('CupertinoPageRoute ignores pointers when user pop gesture is in progress', (WidgetTester tester) async {
+      bool homeTapped = false;
+      await tester.pumpWidget(
+        CupertinoApp(
+          home: TextButton(
+            onPressed: () => homeTapped = true,
+            child: const Text('Page 1'),
+          ),
+        ),
+      );
+
+      tester.state<NavigatorState>(find.byType(Navigator)).push(
+        CupertinoPageRoute<void>(
+          builder: (_) => const Text('Page 2'),
+        ),
+      );
+      await tester.pumpAndSettle();
+
+      expect(find.text('Page 1'), findsNothing);
+
+      final TestGesture swipeGesture = await tester.startGesture(const Offset(5, 100));
+
+      await swipeGesture.moveBy(const Offset(100, 0));
+      await tester.pump();
+
+      expect(find.text('Page 1'), findsOneWidget);
+      expect(tester.state<NavigatorState>(find.byType(Navigator)).userGestureInProgress, true);
+
+      await tester.tap(find.text('Page 1'), warnIfMissed: false);
+      expect(homeTapped, false);
+    });
+
+    testWidgets('CupertinoPageRoute ignores pointers when it is pushed on top of other route', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        CupertinoApp(
+          onGenerateRoute: (_) => CupertinoPageRoute<void>(
+            builder: (_) => const Text('Home'),
+          ),
+        ),
+      );
+
+      await tester.tap(find.text('Home'));
+      tester.state<NavigatorState>(find.byType(Navigator)).push(
+        CupertinoPageRoute<void>(
+          builder: (_) => const Text('Page 2'),
+        ),
+      );
+      await tester.pump();
+      await tester.pump(const Duration(milliseconds: 100));
+
+      expect(find.text('Home'), findsOneWidget); // Transition still in progress
+
+      // Can't test directly for taps because route is interactive but offstage
+      // One ignore pointer for each of two overlay entries (ModalScope, ModalBarrier) on each of two routes
+      expect(find.byType(IgnorePointer, skipOffstage: false), findsNWidgets(4));
+      final List<Element> ignorePointers = find.byType(IgnorePointer, skipOffstage: false).evaluate().toList();
+      expect((ignorePointers.first.widget as IgnorePointer).ignoring, true); // Home modalBarrier
+      expect((ignorePointers[1].widget as IgnorePointer).ignoring, true);    // Home modalScope
+      expect((ignorePointers[2].widget as IgnorePointer).ignoring, true);    // Page 2 modalBarrier
+      expect((ignorePointers.last.widget as IgnorePointer).ignoring, true);  // Page 2 modalScope
```

Comment:
Couldn't come up with a better way of testing this. Ideally you can actually verify that the correct behavior happens when doing a tap, but this doesn't seem to be possible due to how the tester deals with offstage widgets